File: 프로그래머스/lv2/12899. 124 나라의 숫자/124 나라의 숫자.java
class Solution {
    public String solution(int n) {
        StringBuilder sb = new StringBuilder();
        while(n > 0){
            if(n % 3 == 1)
                sb.append(1);
             if(n % 3 == 2)
                sb.append(2);
             if(n % 3 == 0)
                sb.append(4);
            n = n -1;
            n = n / 3;
        }
        return sb.reverse().toString();
    }
}

File: 프로그래머스/lv2/12899. 124 나라의 숫자/124 나라의 숫자.py
def solution(n):
    ans = ''
    while n > 0:
        if n % 3 == 1:
            ans += '1'
        if n % 3 == 2:
            ans += '2'
        if n % 3 == 0:
            ans += '4'
        n = (n-1) // 3
    return ans[::-1]
        
    